content/en/agent/faq/kubernetes-state-cluster-check.md

```diff
@@ -0,0 +1,113 @@
+- link: "agent/autodiscovery/clusterchecks"
+  tag: "Documentation"
+  text: "Running Cluster Checks with Autodiscovery"
+---
+
+The `kubernetes_state` check, which monitors `kube-state-metrics`, is enabled automatically by [Autodiscovery][1]. In larger Kubernetes clusters, this check may consume too much RAM. In this case, consider running the `kubernetes_state` check as a [cluster check][2].
+
+## Configuration
+
+1. Deploy `kube-state-metrics` with cluster check annotations on the service:
+
+	```
+	ad.datadoghq.com/service.check_names: '["kubernetes_state"]'
+	ad.datadoghq.com/service.init_configs: '[{}]'
+	ad.datadoghq.com/service.instances: |
+	    [
+	    {
+	        "kube_state_url": "http://%%host%%:8080/metrics",
+	        "prometheus_timeout": 30,
+	        "min_collection_interval": 30,
+	        "send_pod_phase_service_checks": false,
+	        "telemetry": true
+	    }
+	    ]
+	```
+
+	**Notes**: 8080 is the metrics port. The `prometheus_timeout` and `min_collection_interval` are set to 30 because this setup assumes very large payloads, so these metrics are being collected at a lower granularity. Set `send_pod_phase_service_checks` to `true` if you want the pod phase as a service check. Telemetry is only activated in v4.6.1+ of the Datadog `kube-state-metrics` check.
+
+2. Deploy the Datadog Cluster Agent with the following variables:
+
+	```
+		- name: DD_CLUSTER_CHECKS_ENABLED
+		  value: true
+		- name: DD_EXTRA_CONFIG_PROVIDERS
+		  value: "kube_services"
+		- name: DD_EXTRA_LISTENERS
+		  value: "kube_services" 
+	```          
+
+3. Mount an empty file to override `/etc/datadog-agent/conf.d/kubernetes_state.d/auto_conf.yaml` in the Agent DaemonSet. Otherwise, the Autodiscovered check still runs and may still consume a significant amount of RAM. Configure Agent deployment to mount an empty directory in `/etc/datadog-agent/conf.d/kubernetes_state.d`
+
+	```
+	...
+	    volumeMounts:
+	    ...
+	      - name: empty-dir
+	      mountPath: /etc/datadog-agent/conf.d/kubernetes_state.d
+	    ...
+	  volumes:
+	    - name: empty-dir
+	    emptyDir: {}
+	    ...
+	```
+
+4. Deploy the Agent.
+
+Refer to the [Running Cluster Checks with Autodiscovery][2] documentation for more information. See options that begin with `clusterchecksDeployment` in the Helm chart [README.md][3].
+
+### Verify check dispatching
+
+Run:
+
+```
+kubectl exec -it <datadog-cluster-agent-leader> agent clusterchecks
+```
+
+This command displays something like:
+
+```
+===== Checks on gke-my-cluster-default-pool-b969f074-npsw =====
+
+=== kubernetes_state check ===
+Source: kubernetes-services
+Instance ID: kubernetes_state:4a95df9b407f14d7
+empty_default_hostname: true
+kube_state_url: http://10.59.249.74:8080/metrics
+min_collection_interval: 30
+prometheus_timeout: 30
+send_pod_phase_service_checks: false
+tags:
+- kube_service:kube-state-metrics
+- kube_namespace:kube-system
+- cluster_name:test-cluster
```

Comment:
Offtopic: is there any way to add a custom tag to the list? All our DD dashboards configured to read `kubernetescluster"`tag instead of `cluster_name`.

Reply:
I tried to use the section: https://github.com/helm/charts/blob/master/stable/datadog/values.yaml#L122
But looks it doesn't work, at least in this case when a separate kube-state-metrics pod is running. Also I found that tags value from the Helm chart is translated into this format that looks wrong according to the documentation:

```
    - name: DD_TAGS
      value: '[map[kubernetescluster:k8s-blue-staging]]'
```

Does this looks like a bug?

Reply:
Found working configuration:

```
clusterAgent:
...
  env:
# This also works but it's better not to break the default value 
# but add another one for backward compatibility
# - name: DD_CLUSTER_CHECKS_CLUSTER_TAG_NAME 
# value: "kubernetescluster" 
    - name: DD_CLUSTER_CHECKS_EXTRA_TAGS
      value: "kubernetescluster:my-lovely-cluster"

```